### ISSUE TITLE
[Combobox] Tag list and Combobox interaction inaccessible with screen readers

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2861]
### Provide a general summary of the issue here

The tag list interaction is inaccessible with screen readers, with different screen readers having different experiences. The explanations here are very text heavy, **I will demo this at our meeting and we will discuss this**. A part of the problem is how the tag list itself behaves, and part of the problem is related to the tag list actions when focused on the combobox or listbox. Tag list component problems are mentioned here for now since I've not yet fully tested them and created an issue for it. (TODO: probably redo the whole thing after discussion)

<img width="683" height="90" alt="Multiselectable combobox with Breadcrumbs and Button items selected" src="https://github.com/user-attachments/assets/d707bd97-8397-4dff-a438-78a95db12794" />

### 💁 Possible Solution

This will need a discussion.

If going by the minimum (which in this case is probably the safest option), the tag list function (which is the problematic element here) is redundant for screen reader users at the current form. The combobox itself allows for selection and deselection already. This could probably be hidden from both keyboard and assistive technology users.

I get the idea of the tag list, and it would probably be immensely helpful for many users especially with list boxes that have more than a dozen options. If the tag list function is to be preserved, it needs to be separate from the combobox (i.e., when user is focused on the combobox, the tag list is not operable in any way). This would essentially make these work as two separate functions, which would be way easier to explain to an assistive technology user, and understand as an assistive technology user.

### 🖥️ Steps to Reproduce

**NVDA**
1. Focus on the combobox
2. Select items from the listbox -> items appear as a tag list in front of the combobox/text box
3. Use left or right keys to select items in the tag list

When items are already selected and the combobox is not focused:
1. Press tab -> Focus moves to the tag list, but screen reader stays in browse mode, left or right keys continue reading characters
2. Press tab -> Focus moves to combobox
3. Press shift + tab -> Focus moves back to tab list
4. Press NVDA key (insert or caps lock, depending on selection) + space -> Focus mode activates, user can select items using left or right keys

### What browsers are you seeing the problem on? (only for bugs in code)

Firefox, Chrome, Safari

### What operating system are you using?

Windows, iOS 26.0.1, Android 15

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by anna-lach** [comment]
Notes after the meeting:
- So to make it accessible:
  - Probably we can completely hide for the keyboard users and assistive technology users the tag-list (so keep the tag-list only for the mouse users).
  - It's also possible to use non-removable tags in the tag list (so this will not be focusable, but we will still need to hide it for screen reader users), sth more to decide by @arecuenco-dsgn (UX question).
  - There is also an option to completely move outside the combobox the whole tag-list - it probably can be also accessible (right now it's not because of the fake focus).

- When there are more options, like e.g. 15 items (and more) we should use a version of combobox with `Selected` list.

- Combobox is not a component that should be used for select all / clear all, but only to choose few options from the list.

**Comment by anna-lach** [comment]
We also need to update the documentation, especially with regard to the following issue:

- When there are more options, like e.g. 15 items (and more) we should use a version of combobox with Selected list.

**Comment by Diaan** [comment]
This might be fixed with #2868

**Comment by arecuenco-dsgn** [comment]
After talking with Kacper and Michal, we decided to make these changes:

**Focus Order:**
1. Combobox
<img width="1024" height="163" alt="Image" src="https://github.com/user-attachments/assets/62bb6480-24e4-4c92-9fca-3b3d0ffb16a0" />
2. First tag on the list (close button). Only if there are options selected.
<img width="998" height="137" alt="Image" src="https://github.com/user-attachments/assets/aa1bff73-62cf-45ac-9405-41dfea689474" />

**Things to delete:**
1. Delete the arrow left or right from the dropdown list. 
<img width="1009" height="133" alt="Image" src="https://github.com/user-attachments/assets/8830b698-4eaa-415f-b916-d8f0cda97ead" />

https://github.com/user-attachments/assets/a77cc894-db1f-46ca-8196-5210cd7957d6
